lang/src/org/partiql/lang/syntax/SqlParser.kt

```diff
@@ -2220,6 +2228,23 @@ class SqlParser(private val ion: IonSystem) : Parser {
         return ParseNode(CALL, name, listOf(datePart, timestamp), timestamp.remaining)
     }
 
+    /**
+     * Parses a date string and validates that the date string is a string and of the format YYYY-MM-DD
+     */
+    private fun List<Token>.parseDate(): ParseNode =
+        when (head?.value?.isText) {
+            true -> {
+                // validate that the date string follows the format YYYY-MM-DD
+                try {
+                    LocalDate.parse(head?.value?.stringValue())
+                } catch (e: DateTimeParseException) {
+                    err(e.localizedMessage, PARSE_INVALID_DATE_STRING)
+                }
+                ParseNode(DATE, head, listOf(), tail)
```

Comment:
If we did the needful to parse the date, we definitely should put those fields into the node that we generate.

Reply:
I have modeled it to include the date components.